tests/struct.rs

```diff
@@ -0,0 +1,99 @@
+use serde::{Deserialize, Serialize};
+use serde_bin_prot::to_writer;
+use std::f64::INFINITY;
+
+mod common;
+use common::TestCase;
+
+#[derive(Debug)]
+enum StructTestCases {
+    TestA(A),
+    TestB(B),
+}
+
+#[derive(Serialize, Deserialize, Debug, PartialEq)]
+struct A {
+    x: i64,
+    y: f64,
+}
+
+#[derive(Serialize, Deserialize, Debug, PartialEq)]
+struct B {
+    y: BInner,
+    z: Option<i64>,
+}
+
+#[derive(Serialize, Deserialize, Debug, PartialEq)]
+struct BInner {
+    w: i64,
+    x: i64,
+}
+
+fn struct_test_cases() -> Vec<TestCase<StructTestCases>> {
+    let a0 = A { x: 0, y: 0.0 };
+
+    let a0_expected = vec![0, 0, 0, 0, 0, 0, 0, 0, 0];
```

Comment:
There is cool syntax for array initialization with repeated values
```suggestion
    let a0_expected = vec![0x00; 9];
```

Reply:
nice, forgot about this. updated!